examples/mysql/README.md

```diff
@@ -0,0 +1,97 @@
+# User authentication in MySQL using SVIDs with DNS names
+
+This is an example of how **spiffe-helper** can be used to authenticate users to a **MySQL** database using x509 SVIDs.
+
+MySQL can be compiled using OpenSSL or yaSSL, both of them enable secure connections based on the OpenSSL API.
+In order to handle SVIDs, we need to make sure that the MySQL distribution that we are using was compiled using OpenSSL and not yaSSL.
+
+This statement returns a row if OpenSSL was used and an empty result if yaSSL was used:
+
+```sql
+SHOW STATUS LIKE 'Rsa_public_key';
+```
+
+This guide sets up the authentication configuration for the username: `mysql-user`.
+
+## Guide
+
+### Considerations
+The following assumptions are made:
++ At least one SPIRE server and one agent are deployed with trust domain `example.org`.
+
++ Mysql 8.0 is used in this guide but each step should be easily applicable to other versions as well.
+
+### 1. Install MySQL
+Install [MySQL](https://dev.mysql.com/doc/mysql-installation-excerpt/8.0/en/) and make sure the service is up running.
+```
+service mysql status
+```
+
+### 2. Create the user
+Create the `mysql-user` using the [provided script](create_user.sql).
+```sql
+sudo mysql -u root < examples/mysql/create_user.sql
+```
+
+### 3. Configure MySQL SSL settings
+This guide uses the MySQL default certificates path for simplicity. You may want to create a backup of these files.
+
+```bash
+mv /var/lib/mysql/server-cert.pem /var/lib/mysql/server-cert.pem.bk
+mv /var/lib/mysql/server-key.pem /var/lib/mysql/server-key.pem.bk
+mv /var/lib/mysql/ca.pem /var/lib/mysql/ca.pem.bk
+```
+
+It is also possible to use a different directory. If that case, you should update the MySQL and the spiffe-helper configuration files.
+
+### 4. Start SPIRE server
+Start SPIRE server:
+```bash
+./spire-server run
```

Comment:
in case of UpstreamCA is used it will be added as bundle and generated CA (from spire) is going to be an intermediate, with actual code it will fail, there is a PR to solve it.
remember to add an example without upstream and upstream_bundle with false value, once the solution for that issue is merged we must update this readme to display how to configure to work in both scenaries